### PR TITLE
JavaScript Code Style Linter

### LINKED ISSUE #727: Use a code style linter
Now that we are seeing more contributions, it would be a good time to remove the guesswork from matching our library's style rules. https://github.com/jscs-dev/node-jscs can integrate with our `npm run lint` command, so PRs will catch any issues.

@callmehiphop do you know of a better one than jscs?

### PR DESCRIPTION
**Pull request by callmehiphop**
closes #727 
# To Dos
- [x] Add jscs to `npm run lint` command
- [x] Fix any code that violates style rules
  - [x] `lib/`
  - [x] `system-test/`
  - [x] `test/`
- [x] Get the @stephenplusplus stamp of approval

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"preset": "node-style-guide",
+	"requireTrailingComma": null
+}
```
Comment (stephenplusplus) [inline .jscsrc]: 2 spaces, please! (Ironic to comment on a code style PR about code style).

File: test/datastore/entity.js [hunk 2]
```diff
@@ -24,58 +24,58 @@ var datastore = require('../../lib/datastore');
...
-      'value': {
-          'list_value': [{
-              'integer_value': 6
-          }, {
-              'boolean_value': false
-          }]
-      }
+    name: 'list',
+    value: {
+      list_value: [{
```
Comment (stephenplusplus) [inline test/datastore/entity.js]: Is there a way to force objects be on a new line in an array?:

```js
list_value: [
  {
    //...
  }
]
```

**Comment by stephenplusplus** [comment]
This is so great :sunglasses: :dancer:

File: test/datastore/entity.js (hunk 2 above)
Comment (callmehiphop) [inline test/datastore/entity.js]: Doesn't look like it, there are only a few options for specifically targeting Arrays (as far as I can tell), none of which look to support putting nested objects on a new line.

**Comment by stephenplusplus** [comment]
LGTM. Thanks!

**Pull request merged**